plasmapy/formulary/mathematics.py

```diff
@@ -88,3 +88,77 @@ def Fermi_integral(
         return integral_arr
     else:
         raise TypeError(f"Improper type {type(x)} given for argument x.")
+
+
+def rot_a_to_b(a: np.ndarray, b: np.ndarray) -> np.ndarray:
+    r"""
+    Calculates the 3D rotation matrix that will rotate vector ``a`` to be aligned
+    with vector ``b``. The rotation matrix is calculated as follows. Let
+
+    .. math::
+        \vec v = \vec a \times \vec b
+
+    and let :math:`\theta` be the angle between :math:`\vec a`
+    and :math:`\vec b` such that the projection of :math:`\vec a` along
+    :math:`\vec b` is
+
+    .. math::
+        c = \vec a \cdot \vec b \cos\theta
+
+    Then the rotation matrix :math:`R` is
+
+    .. math::
+        R = I + v_x + v_x^2 \frac{1}{1 + c}
+
+
+    where :math:`I` is the identity matrix and :math:`v_x` is the
+    skew-symmetric cross-product matrix of :math:`v` defined as
+
+    .. math::
+        v_x = \begin{bmatrix}
+                0 & -v_3 & v_2 \\
+                v_3 & 0 & -v_1 \\
+                -v_2 & v_1 & 0
+            \end{bmatrix}
+
+    Note that this algorithm fails when :math:`1+c=0`, which occurs when :math:`a` and
+    :math:`b` are anti-parallel. However, since the correct rotation matrix
+    in this case is simply :math:`R=-I`, this function just handles this
+    special case explicitly.
+
+    This algorithm is based on `this discussion <https://math.stackexchange.com/questions/180418/calculate-rotation-matrix-to-align-vector-a-to-vector-b-in-3d/476311#476311>`_ on StackExchange:
+
+
+    Parameters
+    ----------
+    a : `~numpy.ndarray`, shape (3,)
+        Vector to be rotated.  Should be a 1D, 3-element unit vector.  If ``a``
+        is not normalize, then it will be normalized.
+
+    b : `~numpy.ndarray`, shape (3,)
+        Vector representing the desired orientation after rotation.  Should be
+        a 1D, 3-element unit vector.  If ``b`` is not normalized, then it will
+        be.
+
+    Returns
+    -------
+    R : `~numpy.ndarray`, shape (3,3)
+        The rotation matrix that will rotate vector ``a`` onto vector ``b``.
+
+    """
+
+    # Normalize both vectors
+    a = a / np.linalg.norm(a)
+    b = b / np.linalg.norm(b)
+
+    # Manually handle the case where a and b point in opposite directions
+    if np.dot(a, b) == -1:
```

Comment:
It's not sufficient to rely on `np.dot` to condition `a` and `b` since it will still allow any N-element arrays to pass as long as `a.shape == b.shape`.  The functionality you've coded here only works for 1D, 3-element arrays so you need a workflow like...

```python
a = np.squeeze(a)  # will convert list to array and reduce any single element dimensions
if a.shape != (3,):
    raise ValueError("give message")
a = a / np.linalg.norm(a)
```

Reply:
Ok, done